pkg/api/merge.go

```diff
@@ -0,0 +1,368 @@
+package api
+
+// TODO: needs unit tests
+
+// Notes: shortcomings of the Azure-style API include:
+// 1. There is no mechanism to delete an item from a slice or map without
+// implementing a separate CRUD endpoint.
+// 2. There is no way to set a plain value to its zero value (e.g. bool to
```

Comment:
Isn't this a short-coming of Golang APIs rather than Azure?

Reply:
It's certainly a sort of nasty interaction between the two.  In principle the Golang json package provides access to the underlying parser, which would allow for fixing 2 without going to pointers everywhere, but it's quite a lot of effort.